Library/Homebrew/cmd/upgrade.rb

```diff
@@ -104,11 +106,15 @@ def upgrade
       puts formulae_upgrades.join(", ")
     end
 
-    upgrade_formulae(formulae_to_install)
+    if args.dry_run?
```

Comment:
`return puts "Dry run: did not upgrade anything." if dry_run?` and can skip the `else`